### PR TITLE
Release notes for 0.321.0

### PR DESCRIPTION
**Pull request by bitsondatadev**
## Description
Assemble the release notes for the upcoming 0.321.0 release. Fixes https://github.com/trinodb/trino-python-client/issues/298

## Non-technical explanation
N/A

## Release notes
(x) This is not user-visible or docs only and no release notes are required.

## Verification for each commit
Format: committer/maintainer, PR/issue number, ✅ / ❌ ✅ - release note added and verified, or assessed to be not necessary, set to ❌ before completion

## December 16
@hashhar #252, #27 ✅

## December 22
@hashhar #299 ✅
@hashhar #302 ✅

## December 27
@hashhar #300 ✅

## January 6
@hashhar #311 ✅
@hashhar #307 ✅
@hashhar #305, #308 ✅

## January 11
@hashhar #279, #316 ✅
@hashhar #315 ✅

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -6,6 +6,24 @@ list](https://github.com/trinodb/trino-python-client/tags), the
 [README](https://github.com/trinodb/trino-python-client/blob/master/README.md)
 and the [PyPI page](https://pypi.org/project/trino/).
 
+## Release 0.321.0
+
+* Add support for setting the session timezone. When not set explicitly it
```
Comment (hashhar) [inline CHANGES.md:11]: Move the "Breaking Changes" into their own subheading like in 0.317.0.

**Comment by hashhar** [comment]
I added two more (and re-ordered the list a little bit).

File: CHANGES.md [hunk 2]
```diff
@@ -6,6 +6,43 @@ list](https://github.com/trinodb/trino-python-client/tags), the
 [README](https://github.com/trinodb/trino-python-client/blob/master/README.md)
 and the [PyPI page](https://pypi.org/project/trino/).
 
+## Release 0.321.0
+
+* Add support for SQLAlchemy 2.0.
+  ([#307](https://github.com/trinodb/trino-python-client/issues/307))
+* Add support for `VARBINARY` query parameters.
```
Comment (ebyhr) [inline CHANGES.md:13]: nit: Lowercase type names if we follow the convention in Trino release notes:
```suggestion
* Add support for `varbinary` query parameters.
```

**Review by ebyhr — APPROVED** [review]

**Comment by hashhar** [comment]
I'll merge and do the release later today.

**Pull request merged**